### PR TITLE
AB#7979: Adding Content Hub connector csproj

### PR DESCRIPTION
**Pull request by scottmulligan**
The Content Hub connector now has it's own Foundation project for the base module. The Platform project contains customization built on top of the CH module.

We need to perform transforms in 2 different ways (one for XM Cloud included in the Foundation project and one in the dockerfile for local Docker development).

### CODE REVIEW

File: Sitecore.Demo.Edge.Website.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
```
Comment (scottmulligan) [inline Sitecore.Demo.Edge.Website.sln:3]: @jflheureux - the VS version was updated when saving the solution in VS 2022. It seems to open and build as expected in VS 2019. Let me know if you see any issues.

File: src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.CSharp" />
```
Comment (jflheureux) [inline src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj:46]: All of these references are probably not needed for this project. Could this list please be cleaned? Thanks!

File: src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    <Reference Include="System.Xml" />
+    <Reference Include="System.Configuration" />
+    <Reference Include="System.Web.Services" />
+    <Reference Include="System.EnterpriseServices" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="Properties\AssemblyInfo.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="App_Config\Modules\Sitecore.Connector.CMP\Sitecore.Connector.CMP.Azure.IndexConfiguration.config" />
```
Comment (jflheureux) [inline src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj:68]: In 3 of these 4 configuration files, I see the use of `role:require` with values like `"Standalone or ContentManagement"`. For these configuration sections to be loaded on XM Cloud, we will need to add ` or XMCloud` at the end of these enumerations. Should we do it inside this PR?

**Review by jflheureux — CHANGES_REQUESTED** [review]

File: src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj (hunk 2 above)
Comment (scottmulligan) [reply src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj]: csproj removed

File: src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj (hunk 3 above)
Comment (scottmulligan) [reply src/Foundation/ContentHubConnector/Sitecore.Demo.Edge.Foundation.ContentHubConnector.csproj]: Thank you for noticing this! - it helped me realize there is a lot more going on and I have reworked this PR.

**Comment by jflheureux** [comment]
There is a CH connector Docker assets image added to the CM in the docker-compose.override file as well as in the CM dockerfile. Those would need to be removed too I guess to avoid overriding the base image assets.

**Comment by jflheureux** [comment]
Regarding the `XMCloud` role for the CH connector configs:
1. Have you tested whether it is possible to patch:attribute the "role:require" attribute value to include " or XMCloud" in there?
2. Or we should report it to the XM Cloud team so they fix it in their base image?
3. Or should we still keep a .csproj with a copy of those 3 config files (but with the ` or XMCloud` role:require added) to be deployed on XMCloud?

What do you think?

**Review by jflheureux — APPROVED** [review]

**Pull request merged**